test/framework/integration_framework/fake_peer/behaviour/honest.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/integration_framework/fake_peer/behaviour/honest.hpp"
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/range/adaptor/transformed.hpp>
+#include "backend/protobuf/proto_proposal_factory.hpp"
+#include "backend/protobuf/transaction.hpp"
+#include "common/result.hpp"
+#include "framework/integration_framework/fake_peer/block_storage.hpp"
+#include "framework/integration_framework/fake_peer/network/batches_for_round.hpp"
+#include "framework/integration_framework/fake_peer/proposal_storage.hpp"
+#include "interfaces/iroha_internal/transaction_batch.hpp"
+#include "logger/logger.hpp"
+#include "module/shared_model/builders/protobuf/proposal.hpp"
+#include "validators/default_validator.hpp"
+
+using namespace iroha::expected;
+
+namespace integration_framework {
+  namespace fake_peer {
+
+    HonestBehaviour::HonestBehaviour()
+        : proposal_factory_(
+              std::make_unique<shared_model::proto::ProtoProposalFactory<
+                  shared_model::validation::DefaultProposalValidator>>()) {}
+
+    void HonestBehaviour::processYacMessage(YacMessagePtr message) {
+      getFakePeer().voteForTheSame(message);
+    }
+
+    std::string HonestBehaviour::getName() {
+      return "honest behaviour";
+    }
+
+    LoaderBlockRequestResult HonestBehaviour::processLoaderBlockRequest(
+        LoaderBlockRequest request) {
+      const auto block_storage = getFakePeer().getBlockStorage();
```

Comment:
here because the method signature is
```c++
      boost::optional<const BlockStorage &> getBlockStorage() const;
```
neither below I can see where to make this change.

Reply:
Yes, now I got it, we can assign an rvalue to a const lvalue reference.

Reply:
Yes, now I got it, we can assign an rvalue to a const lvalue reference.